### PR TITLE
Fix #3782 : Made the project use Themes instead of Styles [Dark Mode Implementation].

### LINKED ISSUE #3782: Make the project use Themes instead of Styles [Dark Mode Implementation].
Making the project use themes.xml files instead of existing styles.xml by replacing the existing styles.xml files with themes.xml.

### PR DESCRIPTION
**Pull request by ayush0402**
## Explanation

Fixes #3782
Most part of the existing ```styles.xml``` already followed suggested practices for ```themes.xml``` according to https://medium.com/androiddevelopers/android-styling-themes-vs-styles-ebe05f917578 with semantic naming for color resources.
Separated certain styles suited more for ```themes.xml``` from ```styles.xml```.

Possible solution for further PRs : 
Adding ```themes.xml``` and ```colors.xml``` under ```values-night``` and changing theme's parent to ```DayNight``` variant with changes under ```values-night``` for proper implementation of Dark Mode. 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

**Comment by ayush0402** [comment]
@rt4914 PTAL If this approach is acceptable.

File: app/src/main/res/values-v21/themes.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  </style>
+
+  <style name="AlertDialogTheme" parent="Theme.AppCompat.Light.Dialog.Alert">
+    <item name="colorPrimary">@color/colorPrimary</item>
+    <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
+    <item name="colorAccent">@color/colorPrimary</item>
+    <item name="android:buttonBarNegativeButtonStyle">@style/textAllCaps</item>
+    <item name="android:buttonBarPositiveButtonStyle">@style/textAllCaps</item>
+  </style>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values-v21/themes.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -166,26 +134,20 @@
...
-  </style>
+  <item name="tabIndicatorColor">@color/white</item>
+  <item name="tabIndicatorHeight">4dp</item>
+  <item name="tabPaddingStart">8dp</item>
+  <item name="tabPaddingEnd">8dp</item>
+  <item name="tabBackground">?attr/selectableItemBackground</item>
+  <item name="tabTextAppearance">@style/AppTabTextAppearance</item>
+  <item name="tabIconTint">@color/tab_icon_color_selector</item>
+  <item name="tabSelectedTextColor">@color/white</item>
+</style>
```
Comment (rt4914) [inline app/src/main/res/values/styles.xml]: Fix formatting for this.

File: app/src/main/res/values/themes.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  <style name="SwitchCompatTheme" parent="Theme.MaterialComponents.Light.NoActionBar.Bridge">
+    <item name="colorControlActivated">@color/colorPrimary</item>
+  </style>
+
+  <style name="AlertDialogTheme" parent="Theme.MaterialComponents.Light.Dialog.Alert.Bridge">
+    <item name="colorPrimary">@color/colorPrimary</item>
+    <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
+    <item name="colorAccent">@color/colorPrimary</item>
+  </style>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values/themes.xml]: EOF is missing, add one blank line at the end.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@ayush0402 Nit changes suggested. @BenHenning Can you confirm if this approach is correct.

File: app/src/main/res/values-v21/themes.xml (hunk 1 above)
Comment (ayush0402) [reply app/src/main/res/values-v21/themes.xml]: Done.

File: app/src/main/res/values/themes.xml (hunk 3 above)
Comment (ayush0402) [reply app/src/main/res/values/themes.xml]: Done.

File: scripts/assets/file_content_validation_checks.textproto [hunk 4]
```diff
@@ -47,6 +47,7 @@ file_content_checks {
   exempted_file_name: "app/src/main/res/layout-sw600dp-land/profile_progress_header.xml"
   exempted_file_name: "app/src/main/res/layout-sw600dp-port/profile_progress_header.xml"
   exempted_file_name: "app/src/main/res/values/styles.xml"
+  exempted_file_name: "app/src/main/res/values/themes.xml"
```
Comment (BenHenning) [inline scripts/assets/file_content_validation_checks.textproto]: Which line is failing the check that leads to needing this exemption?

**Review by BenHenning — COMMENTED** [review]
Took a first pass, but I'd like to take a closer look tomorrow. I think this generally looks right, though.

File: scripts/assets/file_content_validation_checks.textproto (hunk 4 above)
Comment (ayush0402) [reply scripts/assets/file_content_validation_checks.textproto]: @BenHenning Sorry. We don't actually need those changes. Reverted it.

File: app/src/main/res/values-v21/themes.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    <item name="windowActionBar">false</item>
+    <item name="windowNoTitle">true</item>
+    <item name="android:windowAnimationStyle">@style/Animation.Activity</item>
+  </style>
+
+  <style name="SwitchCompatTheme" parent="Theme.MaterialComponents.Light.NoActionBar.Bridge">
+    <item name="android:colorControlActivated">@color/colorPrimary</item>
+  </style>
+
+  <style name="AlertDialogTheme" parent="Theme.AppCompat.Light.Dialog.Alert">
```
Comment (BenHenning) [inline app/src/main/res/values-v21/themes.xml]: Should this & above also be named as an Oppia theme @rt4914?

File: app/src/main/res/values/themes.xml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  <style name="OppiaThemeWithoutActionBar" parent="Theme.MaterialComponents.Light.NoActionBar.Bridge">
+    <item name="colorPrimary">@color/colorPrimary</item>
+    <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
+    <item name="colorAccent">@color/colorAccent</item>
+    <item name="windowActionBar">false</item>
+    <item name="windowNoTitle">true</item>
+    <item name="android:windowAnimationStyle">@style/Animation.Activity</item>
+  </style>
+
+  <style name="OppiaThemeNoActionBarColorAccentColorPrimary" parent="Theme.MaterialComponents.Light.NoActionBar.Bridge">
```
Comment (BenHenning) [inline app/src/main/res/values/themes.xml]: Nit on consistency: one theme is 'without actionbar' other is 'no actionbar'.

**Review by BenHenning — COMMENTED** [review]
Thanks @ayush0402. Took a closer look. PR generally LGTM, but had a couple of comments--PTAL.

File: app/src/main/res/values/themes.xml (hunk 6 above)
Comment (ayush0402) [reply app/src/main/res/values/themes.xml]: Done. Renamed NoActionBar to WithoutActionBar.

File: app/src/main/res/values/styles.xml (hunk 2 above)
Comment (ayush0402) [reply app/src/main/res/values/styles.xml]: Done.

File: app/src/main/res/values-v21/themes.xml (hunk 5 above)
Comment (rt4914) [reply app/src/main/res/values-v21/themes.xml]: Makes sense. For consistency we can do that. @ayush0402 PTAL

**Review by rt4914 — APPROVED** [review]
LGTM

File: app/src/main/res/values-v21/themes.xml (hunk 5 above)
Comment (ayush0402) [reply app/src/main/res/values-v21/themes.xml]: Done.

**Comment by ayush0402** [comment]
@BenHenning PTAL.

**Review by BenHenning — APPROVED** [review]
Thanks @ayush0402. This LGTM!

**Pull request merged**